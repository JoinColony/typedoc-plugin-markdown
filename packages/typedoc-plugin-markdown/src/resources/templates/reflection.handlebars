{{> header}}

{{#with model}}

{{#if hasComment}}

{{> comment}}

{{/if}}

{{/with}}

{{#if model.typeParameters}}

## Type parameters

{{#withTypeParameterTable model.typeParameters}}
{{> table }}
{{/withTypeParameterTable}}

{{/if}}

{{#ifShowTypeHierarchy this}}

## Hierarchy

{{#with model.typeHierarchy}}

{{>hierarchy level=0}}

{{/with}}

{{/ifShowTypeHierarchy}}

{{#if model.implementedTypes}}

## Implements

{{#each model.implementedTypes}}
- {{{ type this }}}
{{/each}}

{{/if}}

{{#if model.implementedBy}}

## Implemented by

{{#each model.implementedBy}}
- {{{ type this }}}
{{/each}}

{{/if}}

{{#if model.signatures}}

## Callable

{{#with model}}

{{#each signatures}}

### {{name}}

{{> member.signature showSources=true }}

{{/each}}

{{/with}}

{{/if}}

{{#if model.indexSignature}}

## Indexable

{{#with model}}

{{#with indexSignature}}

{{> index-signature}}

{{/with}}

{{/with}}

{{/if}}

{{#with model}}

{{> main}}

{{/with}}