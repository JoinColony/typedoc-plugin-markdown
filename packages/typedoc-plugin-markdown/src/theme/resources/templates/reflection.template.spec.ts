import { context, project } from '../../../../test/test-utils';
import { reflectionTemplate } from './reflection.template';

describe(`Reflection Template:`, () => {
  beforeEach(async () => {});

  test(`should compile project reflection`, () => {
    const props = {
      project: project(),
      model: project(),
    } as any;
    expect(reflectionTemplate(context(), props)).toMatchSnapshot();
  });

  test(`should compile callable reflection`, () => {
    const props = {
      project: project(),
      model: project().findReflectionByName('CallableReflection'),
    } as any;
    expect(reflectionTemplate(context(), props)).toMatchSnapshot();
  });

  test(`should compile indexable reflection`, () => {
    const props = {
      project: project(),
      model: project().findReflectionByName('IndexableReflection'),
    } as any;
    expect(reflectionTemplate(context(), props)).toMatchSnapshot();
  });

  test(`should compile implemented class reflection`, () => {
    const props = {
      project: project(),
      model: project().findReflectionByName('ImplementedClass'),
    } as any;
    expect(reflectionTemplate(context(), props)).toMatchSnapshot();
  });

  test(`should compile enum reflection`, () => {
    const props = {
      project: project(),
      model: project().findReflectionByName('EnumReflection'),
    } as any;
    expect(reflectionTemplate(context(), props)).toMatchSnapshot();
  });
});
