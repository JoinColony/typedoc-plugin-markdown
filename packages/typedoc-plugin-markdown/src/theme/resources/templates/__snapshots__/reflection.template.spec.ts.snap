// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reflection Template: should compile callable reflection 1`] = `
"[breadcrumbsPartial]

# [reflectionTitlePartial]

[reflectionPathPartial]

## Callable

### CallableReflection

[reflectionTitlePartial]

[tocPartial]"
`;

exports[`Reflection Template: should compile enum reflection 1`] = `
"[breadcrumbsPartial]

# [reflectionTitlePartial]

[reflectionPathPartial]

[tocPartial]

[groupsPartial]"
`;

exports[`Reflection Template: should compile implemented class reflection 1`] = `
"[breadcrumbsPartial]

# [reflectionTitlePartial]

[reflectionPathPartial]

## Implements

- [typePartial]

[tocPartial]

[groupsPartial]"
`;

exports[`Reflection Template: should compile indexable reflection 1`] = `
"[breadcrumbsPartial]

# [reflectionTitlePartial]

[reflectionPathPartial]

## Indexable

[indexSignatureTitlePartial]

[tocPartial]"
`;

exports[`Reflection Template: should compile project reflection 1`] = `
"[breadcrumbsPartial]

# [reflectionTitlePartial]

[reflectionPathPartial]

[tocPartial]

[groupsPartial]"
`;
